lib/processors/minifier.js

```diff
@@ -1,21 +1,44 @@
+import path from "node:path";
+import {fileURLToPath} from "node:url";
 import posixPath from "node:path/posix";
-import {minify} from "terser";
+import os from "node:os";
+import workerpool from "workerpool";
 import Resource from "@ui5/fs/Resource";
+import {getLogger} from "@ui5/logger";
+const log = getLogger("builder:processors:minifier");
 
-/**
- * Preserve comments which contain:
- * <ul>
- *  <li>copyright notice</li>
- *  <li>license terms</li>
- *  <li>"@ui5-bundle"</li>
- *  <li>"@ui5-bundle-raw-include"</li>
- * </ul>
- *
- * @type {RegExp}
- */
-const copyrightCommentsAndBundleCommentPattern = /copyright|\(c\)(?:[0-9]+|\s+[0-9A-Za-z])|released under|license|\u00a9|^@ui5-bundle-raw-include |^@ui5-bundle /i;
 const debugFileRegex = /((?:\.view|\.fragment|\.controller|\.designtime|\.support)?\.js)$/;
 
+const MIN_WORKERS = 2;
+const MAX_WORKERS = 4;
+const osCpus = os.cpus().length || 1;
+const maxWorkers = Math.max(Math.min(osCpus - 1, MAX_WORKERS), MIN_WORKERS);
+
+// Shared workerpool across all executions until the taskUtil cleanup is triggered
+let pool;
+
+function getPool(taskUtil) {
+	if (!pool) {
+		log.verbose(`Creating workerpool with up to ${maxWorkers} workers (available CPU cores: ${osCpus})`);
+		const workerPath = path.join(path.dirname(fileURLToPath(import.meta.url)), "minifierWorker.js");
```

Comment:
You could avoid the import of node:path by using the native URL API:

 const workerPath = fileURLToPath(new URL("./minifierworker.js", import.meta.url));

Reply:
Thank you very much. I didn't take the time to look into the URL mechanisms for file paths yet. Done and tested 👍